File: sandbox-runner/patterns/mapreduce-tree/java/src/main/java/Shared.java
import java.util.ArrayList;
import java.util.List;

public final class Shared {
    public static final String TASK_QUEUE = "mapreduce-tree-task-queue";
    public static final String WORKFLOW_ID_PREFIX = "mapreduce-tree";
    public static final int LEAF_THRESHOLD = 3;
    public static final int MAX_DEPTH = 5;
    public static final String RESULT_SIGNAL = "nodeResult";

    public static final List<String> RECORDS;

    static {
        RECORDS = new ArrayList<>();
        for (int i = 0; i < 9; i++) {
            RECORDS.add("item-" + i);
        }
    }

    private Shared() {}
}

File: sandbox-runner/patterns/mapreduce-tree/go/shared.go
package main

import "fmt"

const (
	TaskQueue        = "mapreduce-tree-task-queue"
	WorkflowIDPrefix = "mapreduce-tree"
	LeafThreshold    = 3
	MaxDepth         = 5
	ResultSignal     = "nodeResult"
)

// Records is the demo record set.
var Records = func() []string {
	r := make([]string, 9)
	for i := range r {
		r[i] = fmt.Sprintf("item-%d", i)
	}
	return r
}()

// ResultPayload is the signal payload sent from child to parent.
type ResultPayload struct {
	ID      string   `json:"id"`
	Results []string `json:"results"`
}
